react/headless/src/utils/stores.ts

```diff
@@ -16,9 +16,19 @@ export function useObservable<T>(store$: ReadableSignal<T>) {
 
 	useEffect(() => {
 		return store$.subscribe((arg) => {
+			console.log('in use observable', arg);
 			// We're passing an update function to setValue here instead of just doing setValue(arg) so that
 			// if arg is a function, it is properly used as the value instead of being called as an update function
-			setValue(() => arg);
+			let val = arg;
+			//React does a comparison of new state and old state and in this case (using Object.is), it compares the references of new Map/Set to old Map/Set which share the same value.
+			//So, React bails out without triggering an update. A new Map/Set is created in this case.
+			if (arg instanceof Map) {
```

Comment:
__OPEN QUESTION__
Here we should forward every emission received from the tansu store, as this one is handling the equality of the objects. unfortunately I could not find a way to bypass the react default equality when doing a 'setState' or at least passing a function which returns not equal always. 
So I came up with an workaround, but this is not scalable and we do more computation than necessary.
Any ideas ?